### PR TITLE
Add `dials`

### PR DESCRIPTION
**Pull request by ndevenish**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by Anthchirp** [comment]
Happy to be a maintainer

**Comment by rjgildea** [comment]
Happy to be added as a maintainer 👍

File: recipes/dials/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    - {{ pin_compatible('boost') }}
+    - boost-cpp
+    - {{ pin_compatible('cctbx-base') }}
+    - cctbx
+    - {{ pin_compatible('dxtbx') }}
+    - gemmi
+    - h5py
+    - jinja2
+    - matplotlib-base
+    - {{ pin_compatible('numpy') }}
```
Comment (Anthchirp) [inline recipes/dials/meta.yaml:40]: Is this compiling against numpy or just using numpy?
If it is compiling against, then should add numpy to host dependencies for completeness sake.

Also alphabetical sorting of dependencies wouldn't go amiss 😉

**Review by Anthchirp — APPROVED** [review]

File: recipes/dials/meta.yaml (hunk 1 above)
Comment (Anthchirp) [reply recipes/dials/meta.yaml:40]: (And if only importing then just list it as `numpy`)

Comment (ndevenish) [reply recipes/dials/meta.yaml:40]: That's a a good point, I don't think dials uses it at compile time, only dxtbx and cctbx. I think I left the pin_compatible _for_ these dependencies, but imagine that their restrictions would work.

Also, I'd tried to make it package-alphabetical but it's probably cleaner to do it in a way that can be simply sorted.

**Comment by Anthchirp** [comment]
Looks good to me

**Pull request merged**